### ISSUE TITLE
AWS Request signer seems broken

### ISSUE THREAD

**Issue by iAndThen** [taskrabbit/elasticsearch-dump#411]
- ElasticDump version -- 3.3.7
- Elasticsearch version -- 5.5
- Node.js version -- 8.1

Input:
```
elasticdump \
  --input=<aws_host>/index \
  --output=data.json \
  --type=data \
  --awsChain
```
Produces output:
```
Wed, 04 Apr 2018 10:19:09 GMT | starting dump
Wed, 04 Apr 2018 10:19:10 GMT | got 100 objects from source elasticsearch (offset: 0)
Wed, 04 Apr 2018 10:19:10 GMT | sent 100 objects to destination file, wrote 100
Wed, 04 Apr 2018 10:19:10 GMT | Error Emitted => {"message":"The request signature we calculated does not match the signature you provided. Check your AWS Secret Access Key and signing method. Consult the service documentation for details.\n\nThe Canonical String for this request should have been\n'GET\n/_search/scroll\n\ncontent-length:259\ncontent-type:application/json\nhost:<AWS_HOST>\nx-amz-date:20180404T101910Z\n\ncontent-length;content-type;host;x-amz-date\ncff198a04f728fa5be43e54c3de98dac14a91a2c59c8be820a74d43d48a8028e'\n\nThe String-to-Sign should have been\n'AWS4-HMAC-SHA256\n20180404T101910Z\n20180404/eu-west-1/es/aws4_request\<DATA>'\n"}
Wed, 04 Apr 2018 10:19:10 GMT | Total Writes: 100
Wed, 04 Apr 2018 10:19:10 GMT | dump ended with error (get phase) => Error: {"message":"The request signature we calculated does not match the signature you provided. Check your AWS Secret Access Key and signing method. Consult the service documentation for details.\n\nThe Canonical String for this request should have been\n'GET\n/_search/scroll\n\ncontent-length:259\ncontent-type:application/json\nhost:<AWS_HOST>\nx-amz-date:20180404T101910Z\n\ncontent-length;content-type;host;x-amz-date\ncff198a04f728fa5be43e54c3de98dac14a91a2c59c8be820a74d43d48a8028e'\n\nThe String-to-Sign should have been\n'AWS4-HMAC-SHA256\n20180404T101910Z\n20180404/eu-west-1/es/aws4_request\<DATA>'\n"}
```

Then I switched to the previous version of Elasticdump I used before (3.3.1) and the same command works fine. 

I suspect that AWS Request Signer was changed or even broken. 
In any case, it is impossible to work with AWS ES using Elasticdump.

**Comment by ferronrsmith** [comment]
The only change made to AWS ES was the addition of the content-type headerr

**Comment by ferronrsmith** [comment]
I remember that change, but it's strange that you have the issue.

**Comment by iAndThen** [comment]
I use 3.3.4
But using version 3.3.5+ is not possible for AWS users.

**Comment by ferronrsmith** [comment]
I've added a fallback for aws to the `chore/aws` branch. Can you test it and let me know ?

**Comment by iAndThen** [comment]
Tested with AWS ES 5.5
Works fine.

**Comment by ferronrsmith** [comment]
ok great thanks. I'll make a release

**Comment by iAndThen** [comment]
Thanks!

**Comment by ferronrsmith** [comment]
npm publish is down. I will push as soon as they have resolved the issue

**Comment by ferronrsmith** [comment]
ok published

**Comment by farhansalam** [comment]
@ferronrsmith have you released it yet? having same issue!

**Comment by ferronrsmith** [comment]
yes this has been released since `3.3.8`. What's the query you are trying to execute ?
